presto-docs/src/main/sphinx/security/hive.rst

```diff
@@ -0,0 +1,241 @@
+                                                   When using the default value of ``SIMPLE``, Kerberos
+                                                   authentication is disabled and no other properties need to
+                                                   be configured.
+
+``hive.hdfs.presto.principal``                     The Kerberos principal that Presto will use when connecting
+                                                   to HDFS.
+
+``hive.hdfs.presto.keytab``                        The path to the keytab file that contains a key for the
+                                                   principal specified by ``hive.hdfs.presto.principal``.
+
+================================================== ============================================================ ==========
+
+.. _hive-security-simple:
+
+``SIMPLE``
+^^^^^^^^^^
+
+.. code-block:: none
+
+    hive.hdfs.authentication.type=SIMPLE
+
+``SIMPLE`` authentication is the default authentication mechanism for HDFS.
+When using ``SIMPLE`` authentication, Presto connects to HDFS using Hadoop's
+simple authentication mechanism. Kerberos is not used.
+
+.. _hive-security-kerberos:
+
+``KERBEROS``
+^^^^^^^^^^^^
+
+.. code-block:: none
+
+    hive.hdfs.authentication.type=KERBEROS
+    hive.hdfs.presto.principal=hdfs@EXAMPLE.COM
+    hive.hdfs.presto.keytab=/etc/presto/hdfs.keytab
+
+When using ``KERBEROS`` authentication, Presto accesses HDFS as the principal
+specified by the ``hive.hdfs.presto.principal`` property. Presto will
+authenticate this principal using the keytab specified by the
+``hive.hdfs.presto.keytab`` keytab.
+
+Keytab files must be distributed to every node in the cluster that runs Presto.
+
+:ref:`Additional information on keytab files.<hive-security-additional-keytab>`
+
+.. _hive-security-impersonation:
+
+End User Impersonation
+======================
+
+Impersonation Accessing HDFS
+----------------------------
+
+Presto can impersonate the end user who is running a query. In the case of a
+user running a query from the command line interface, this is the value of the
+username associated with the Presto cli process or the value of the optional
+``--user`` option.  Impersonating the end user can provide additional security
+when accessing HDFS if HDFS permissions or ACLs are used.
+
+.. _hive-security-simple-impersonation:
+
+``SIMPLE`` authentication with HDFS impersonation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: none
+
+    hive.hdfs.authentication.type=SIMPLE
+    hive.hdfs.impersonation=true
+
+When using ``SIMPLE`` authentication with impersonation, Presto impersonates
+the user who is running the query when accessing HDFS. The user Presto is
+running as must be allowed to impersonate this user, as discussed in the
+section :ref:`configuring-hadoop-impersonation`.
+
+.. _hive-security-kerberos-impersonation:
+
+``KERBEROS`` authentication with HDFS impersonation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
Just noticed this. For topics at this level, upper case (title format) isused in titles in Teradata docs. But if all Presto documentation uses this format for topics at this level, this should be kept as is...